src/main/java/UI/Parser.java

```diff
@@ -0,0 +1,142 @@
+        String arguments = (spaceIndex != -1) ? input.substring(spaceIndex + 1).trim() : "";
+
+        switch (command) {
+        case "mark":
+        case "unmark":
+            handleMarkUnmark(command, arguments);
+            break;
+        case "todo":
+            handleToDo(arguments);
+            break;
+        case "deadline":
+            handleDeadline(arguments);
+            break;
+        case "event":
+            handleEvent(arguments);
+            break;
+        case "list":
+            handleList();
+            break;
+        case "delete":
+            handleDelete(arguments);
+            break;
+        case "bye":
+            handleBye();
+            return false;
+        case "find":
+            handleFind(arguments);
+            break;
+        default:
+            throw new BotException("I'm sorry, I don't recognize that command.");
+        }
+        return true;
+    }
+
+    private static void handleFind(String arguments) throws BotException{
+        if (arguments.isEmpty()) throw new BotException("Please provide a search term");
+        TaskList searchResultList = TaskList.mainTaskList.tasksContainingTerm(arguments);
+        searchResultList.printList();
+        if (searchResultList.getNumTasks() == 1) {
+            System.out.printf("There is 1 task containing the word \"%s\".\n", arguments);
+        } else {
+            System.out.printf("There are %s tasks containing the word \"%s\".\n", searchResultList.getNumTasks(), arguments);
+        }
+    }
+
+    private static void handleMarkUnmark(String command, String arguments) throws BotException {
+        if (arguments.isEmpty()) throw new BotException("Please provide a task number.");
+        int taskIndex = Integer.parseInt(arguments) - 1;
+        if (taskIndex < 0 || taskIndex >= TaskList.mainTaskList.getNumTasks()) throw new BotException("That task does not exist!");
+
+        if (command.equals("mark")) {
+            TaskList.mainTaskList.markTask(taskIndex);
+            System.out.println("Nicely done! Keep it up!\n");
+        } else {
+            TaskList.mainTaskList.unmarkTask(taskIndex);
+            System.out.println("Sure, I'll uncheck that task!\n");
+        }
+        Parser.handleList();
+    }
+
+    private static void handleToDo(String arguments) throws BotException {
+        if (arguments.isEmpty()) throw new BotException("Please provide a task description.");
+        new ToDo(arguments);
+        UI.taskAddedMsg();
+    }
+
+    private static void handleDeadline(String arguments) throws BotException {
+        int byIndex = arguments.indexOf(" /by ");
+        if (byIndex == -1) throw new BotException("Please format your instructions correctly. E.g., deadline [task] /by [MMddyyyy HHmm]");
+        String taskDescription = arguments.substring(0, byIndex);
+        String deadline = arguments.substring(byIndex + 5);
```

Comment:
Have you considered:
```suggestion
        String deadline = arguments.split(" /by ")[1].trim();
        if (deadline.isEmpty()) {
            throw new BotException("Please format your instructions correctly. E.g., deadline [task] /by [MMddyyyy HHmm]");
        }
```

Reply:
This would throw an invalid index error if there was no " /by " but I could just wrap it with a try block. Your suggestions looks a lot more concise thanks!